product_state/models/product_state.py

```diff
@@ -42,3 +44,14 @@ def _compute_products_count(self):
         }
         for state in self:
             state.products_count = mapped_data.get(state.id, 0)
+
+    @api.constrains("default")
+    def constrains_default_state(self):
+        rec = self.search([("default", "=", True)])
+        if len(rec) != 1:
+            raise ValidationError(
+                _(
+                    "There should be only one default state, current default state: '%s'"
+                    % (rec - self).name
+                )
+            )
```

Comment:
```suggestion
    def _check_default(self):
        if self.search_count([("default", "=", True)]) > 1:
            raise ValidationError(
                _(
                    "There should be only one default state, current default state: '%s'"  
                )
            )
```

Reply:
I think we should let user know which state is default